src/main/java/com/example/demo/rest/TaskResource.java

```diff
@@ -31,15 +34,19 @@ public class TaskResource {
     UserInfoDao userInfoDao;
 
     @GetMapping
-    public ResponseEntity<?> getUserTasks(@PathVariable("userId") int userId, @RequestParam("groupId") Integer groupId) {
+    public ResponseEntity<?> getUserTasks(@PathVariable("userId") int userId, @RequestParam("groupId") Integer groupId,
+                                          @RequestParam("toDate") @DateTimeFormat(pattern = "yyyy-MM-dd") java.util.Date toDate,
+                                          @RequestParam("fromDate")@DateTimeFormat(pattern = "yyyy-MM-dd") java.util.Date fromDate) {
         Optional<UserInfo> user = userInfoDao.findById(userId);
         if (!user.isPresent()) {
             return new ResponseEntity<>(HttpStatus.NOT_FOUND);
         }
-        if (groupId != null) {
-            return new ResponseEntity<>(user.get().getUserWithTasks(), HttpStatus.OK);
+        if (groupId == null) {
+            return new ResponseEntity<>(user.get().getUserWithTasks().getTasks().stream().filter(task -> task.getCreatedDate().after(fromDate)
+                    && task.getCreatedDate().before(toDate)).collect(Collectors.toList()), HttpStatus.OK);
```

Comment:
Can we try to apply the date filtering in the SQL itself rather than fetching all records and filtering in java?